pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -479,6 +487,24 @@ func (vm *volumeManager) WaitForUnmount(ctx context.Context, pod *v1.Pod) error
 	return nil
 }
 
+func (vm *volumeManager) WaitForAllPodsUnmount(ctx context.Context, pods []*v1.Pod) error {
+	var (
+		errors []error
+		wg     sync.WaitGroup
+	)
+	wg.Add(len(pods))
+	for _, pod := range pods {
+		go func(pod *v1.Pod) {
+			defer wg.Done()
+			if err := vm.WaitForUnmount(ctx, pod); err != nil {
```

Comment:
if `WaitForUnmount` times out - does it return error or exist with `nil`?

Reply:
If `WaitForUnmount` times out and there are still volumes mounted, it returns an error. If the function times out but all volumes have been unmounted it returns `nil`:

https://github.com/kubernetes/kubernetes/blob/dd4943c831dbc43fb8d410ccdbc27e3c807131fa/pkg/kubelet/volumemanager/volume_manager.go#L468-L470